kafka/protocol/message.py

```diff
@@ -154,20 +154,24 @@ class MessageSet(AbstractType):
     HEADER_SIZE = 12  # offset + message_size
 
     @classmethod
-    def encode(cls, items):
+    def encode(cls, items, prepend_size=True):
         # RecordAccumulator encodes messagesets internally
         if isinstance(items, (io.BytesIO, KafkaBytes)):
             size = Int32.decode(items)
-            # rewind and return all the bytes
-            items.seek(items.tell() - 4)
+            if prepend_size:
```

Comment:
I think you also need to alter the number of bytes read depending on whether prepend_size is True / False. Below it is reading size + 4, but that is only correct if we are prepending size. If we are not, I think we only want to read size bytes?

Reply:
Yup, seems like I missed it...